examples/prom/nonlinear_elasticity_global_rom.cpp

```diff
@@ -1691,16 +1912,812 @@ void RomOperator::Mult_FullOrder(const Vector& vx, Vector& dvx_dt) const
+    add(*Vx_temp, *x0, Vx);
+    P->Mult(Vx, p_Vx);
+
+    // Initialize nonlinear operator storage
+    // Assuming all elements have the same dim and n_dof
+    fe = fesR->GetFE(0);
+    dof = fe->GetDof();
+    dim = fe->GetDim();
+    int index = 0;
+    DenseMatrix DS(dof, dim);
+    DenseMatrix Jpt(dim);
+    DenseMatrix P_f(dim);
+    DenseMatrix PMatI; // Extract element dofs
+    DenseMatrix PMatO;
+    Vector elvect(dof * dim);
+    PMatO.UseExternalData(elvect.GetData(), dof, dim);
+
+    eprev = -1;
+    double temp = 0.0;
+
+    // For every quadrature weight
+    for (int i = 0; i < qp.size(); ++i) // NOTE: i < 9
+    {
+        const int e = qp[i] / nqe; // Element index
+        // Local (element) index of the quadrature point
+        const int qpi = qp[i] - (e * nqe);
+        const IntegrationPoint &ip = ir->IntPoint(qpi);
+
+        if (e != eprev) // Update element transformation
+        {
+            doftrans = fesR->GetElementVDofs(e, vdofs);
+            fe = fesR->GetFE(e);
+            eltrans = fesR->GetElementTransformation(e);
+
+            dof = fe->GetDof(); // Get number of dofs in element
+            dim = fe->GetDim();
+
+            if (doftrans)
+            {
+                MFEM_ABORT("TODO: Implementation for when `doftrans` is not NULL");
+            }
+
+            // Get element vectors
+            p_Vx.GetSubVector(vdofs, Vx_e);
+            eprev = e;
+        }
+
+        // Integration at ip
+        elvect = 0.0;
+        PMatI.UseExternalData(Vx_e.GetData(), dof, dim);
+
+        // Set integration point in the element transformation
+        eltrans->SetIntPoint(&ip);
+        model->SetTransformation(*eltrans);
+
+        for (int ii = 0; ii < dof; ++ii)
+        {
+            for (int jj = 0; jj < dim; ++jj)
+            {
+                index = jj + ii * dim;
+                DS.Elem(ii, jj) = DS_coef[index + (i * elvect.Size())];
+            }
+        }
+
+        MultAtB(PMatI, DS, Jpt);
+        model->EvalP(Jpt, P_f);
+        AddMultABt(DS, P_f, PMatO);
+
+        // Calculate r[i] = ve_j^T * elvect
+        // coef is size len(vdofs) * rvdim * rw.size
+        for (int j = 0; j < rvdim; ++j)
+        {
+            temp = 0.0;
+            for (int k = 0; k < elvect.Size(); k++)
+            {
+                temp += coef[k + (i * elvect.Size()) + (j * qp.size() * elvect.Size())] *
+                        elvect[k];
+            }
+            res[j] += temp;
+        }
+    }
+}
+
+void ComputeElementRowOfG(const IntegrationRule *ir, Array<int> const &vdofs,
+                          Vector const &ve_j, NeoHookeanModel *model, Vector const &elfun,
+                          FiniteElement const &fe, ElementTransformation &Trans, Vector &r)
+{
+    MFEM_VERIFY(r.Size() == ir->GetNPoints(), "");
+    int dof = fe.GetDof(); // Get number of dofs in element
+    int dim = fe.GetDim();
+
+    // Initialize nonlinear operator matrices (there is probably a better way)
```

Comment:
To avoid allocating these matrices on every call to this function, this can become a struct or class.

Reply:
That has been implemented in the latest commit!